grpc/stream.go

```diff
@@ -284,6 +279,17 @@ func (s *stream) writeData(wg *sync.WaitGroup) {
 	}
 }
 
+func (s *stream) processSendError(err error) {
+	if errors.Is(err, io.EOF) {
+		s.logger.WithError(err).Debug("skip sending a message stream is cancelled/finished")
+		err = nil
+	}
```

Comment:
I would still prefer to just skip that ... 

We can't know if this EOF is because that was what was expected.

If it isn't this is just shuffing it under the rug.
If it is - I feel the user should fix that. Maybe we can *not* emit it if the WebSocket (js object) is already closed, but doesn't seem like what the user reported to me.

Reply:
>If it isn't this is just shuffing it under the rug.

But according to [to this](https://github.com/grpc/grpc-go/blob/master/stream.go#L866-L870), the real error will be returned with the `RecvMsg`. So EOF in sending context, is not the real error.

However, I now I'm thinking that we should not do `closeWithError(nil)` for that case :thinking: since it could hijack the real error reporting. :thinking:

Reply:
>However, I now I'm thinking that we should not do closeWithError(nil)

In principle, it sounds correct to me

 >since it could hijack the real error reporting

but I'm not getting how this is possible. It sounds like `closeWithError` does not affect directly the read loop which is the place where it is expected to get the real error as you reported.

Reply:
>Or is it for the fact that closeWithError is called twice but the second is discarded?

yeah, at the current state, `closeWithError` is designed to discard the second error if the stream is closed. So I think that I need to re-think this and maybe split the responsibilities to actually about the closing and error reporting.

Reply:
>So I think that I need to re-think this and maybe split the responsibilities to actually about the closing and error reporting.

What is the expectation, will it land in this PR?

Reply:
Okay, I've pushed the commit where tried to address the issue that I've described above:

>However, I now I'm thinking that we should not do closeWithError(nil) for that case thinking since it could hijack the real error reporting. thinking

Regarding the omitting the EOF (in sending), I still believe that's the right thing since as I linked to the grpc's code the real error if that happens is routing to the `RecvMsg`